iFixit/react-commerce
# Implement pro pricing on product page

closes #689 

## Changelog

- Introduces support for pro prices in the product page and in the drawer
- Refactors some price related dependencies into monorepo packages so that they can be used by the cart drawer
- Refactors the cart sdk to abstract away cart api implementation details and provide a consistent interface to manipulate the cart line items

## Known issues

- We noticed that the cart api returns different pro prices. Most likely this is due to data being out of sync in the product variant metafield cc @sterlinghirsh 
- https://github.com/iFixit/ifixit/issues/44855

## QA

1. Visit Vercel preview
2. Login on cominor as a pro user
3. Visit a [product page](https://react-commerce-git-add-product-page-pro-pricing-ifixit.vercel.app/Products/pro-tech-toolkit)
4. Verify that discounted prices have the pro badge (check the product price, the cross sell items and the featured products)
5. Add something to the cart and verify that the cart drawer also displays pro prices
6. Play with the cart drawer and verify that it keeps working correctly
7. Try to checkout and verify that you are correctly redirected to the Shopify checkout
8. Repeat previous step but logged out (or logged as a non-pro user)
